### PR TITLE
Dynamically resolve available db engine versions

### PR DESCRIPTION
**Pull request by zliang-akamai**
## 📝 Description

DBaaS upgrade led to failure tests in this repo last time because the db engine versions we used were no longer supported. A version update PR solved that but ideally, we would want to dynamically get the available engine versions and apply them to the tests to prevent this issue permanently.

## ✔️ How to Test

`make TEST_ARGS="-v mysql_basic" test`
`make TEST_ARGS="-v mysql_complex" test`

`make TEST_ARGS="-v postgresql_basic" test`
`make TEST_ARGS="-v postgresql_complex" test`

### CODE REVIEW

**Comment by zliang-akamai** [comment]
/acctest tests=firewall_list sha=1d289e9ec9ea72b814bee319e08a5fdeff2eb2f7

**Comment by zliang-akamai** [comment]
/acctest tests= mysql_basic sha=1d289e9ec9ea72b814bee319e08a5fdeff2eb2f7

**Comment by zliang-akamai** [comment]
/acctest tests= mysql_complex sha=1d289e9ec9ea72b814bee319e08a5fdeff2eb2f7

**Comment by zliang-akamai** [comment]
/acctest tests= postgresql_basic sha=1d289e9ec9ea72b814bee319e08a5fdeff2eb2f7

**Comment by zliang-akamai** [comment]
/acctest tests= postgresql_complex sha=1d289e9ec9ea72b814bee319e08a5fdeff2eb2f7

**Comment by zliang-akamai** [comment]
/acctest tests= mysql_basic sha=83624d90991e1678f92a3d8ea0387f821fbca7e5

**Comment by zliang-akamai** [comment]
/acctest tests=mysql_basic sha=83624d90991e1678f92a3d8ea0387f821fbca7e5

**Comment by zliang-akamai** [comment]
/acctest tests=mysql_complex sha=83624d90991e1678f92a3d8ea0387f821fbca7e5

**Comment by zliang-akamai** [comment]
/acctest tests=postgresql_basic sha=83624d90991e1678f92a3d8ea0387f821fbca7e5

**Comment by zliang-akamai** [comment]
/acctest tests= postgresql_complex sha=83624d90991e1678f92a3d8ea0387f821fbca7e5

**Comment by zliang-akamai** [comment]
/acctest tests=postgresql_complex sha=83624d90991e1678f92a3d8ea0387f821fbca7e5

**Comment by zliang-akamai** [comment]
/acctest tests=mysql_basic sha=67373d62b8678d49b0fff67225750d414feb1ccc

**Comment by zliang-akamai** [comment]
/acctest tests=mysql_complex sha=67373d62b8678d49b0fff67225750d414feb1ccc

**Comment by zliang-akamai** [comment]
/acctest tests=postgresql_basic sha=67373d62b8678d49b0fff67225750d414feb1ccc

**Comment by zliang-akamai** [comment]
/acctest tests=postgresql_complex sha=67373d62b8678d49b0fff67225750d414feb1ccc

**Review by lgarber-akamai — APPROVED** [review]
LGTM! Nice work

File: tests/integration/targets/postgresql_complex/tasks/main.yaml [hunk 1]
```diff
@@ -42,7 +57,7 @@
           - db_create.database.allow_list | length == 1
           - db_create.database.allow_list[0] == '0.0.0.0/0'
           - db_create.database.engine == 'postgresql'
-          - db_create.database.version == '14.6'
+          - db_create.database.version == ''{{ engine_version }}''
```
Comment (lgarber-akamai) [inline tests/integration/targets/postgresql_complex/tasks/main.yaml:60]: Looks like there's an extra quote here 👍

**Comment by zliang-akamai** [comment]
/acctest tests=mysql_basic sha=67373d62b8678d49b0fff67225750d414feb1ccc

**Comment by zliang-akamai** [comment]
/acctest tests=mysql_complex sha=67373d62b8678d49b0fff67225750d414feb1ccc

**Comment by zliang-akamai** [comment]
/acctest tests=postgresql_basic sha=942843b216749cbe23e11f1eaa619a3831607d01

**Comment by zliang-akamai** [comment]
/acctest tests=postgresql_complex sha=942843b216749cbe23e11f1eaa619a3831607d01

**Comment by zliang-akamai** [comment]
/acctest tests=mysql_basic sha=942843b216749cbe23e11f1eaa619a3831607d01

**Comment by zliang-akamai** [comment]
/acctest tests=mysql_complex sha=942843b216749cbe23e11f1eaa619a3831607d01

File: tests/integration/targets/postgresql_complex/tasks/main.yaml (hunk 1 above)
Comment (zliang-akamai) [reply tests/integration/targets/postgresql_complex/tasks/main.yaml]: Oops

**Review by jriddle-linode — APPROVED** [review]

**Pull request merged**